src/native-stack/utils/useBackPressSubscription.tsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import { BackHandler, NativeEventSubscription } from 'react-native';
+
+interface Args {
+  onBackPress: () => boolean;
+  isDisabled: boolean;
+}
+
+interface UseBackPressSubscription {
+  handleAttached: () => void;
+  handleDetached: () => void;
+  createSubscription: () => void;
+  clearSubscription: () => void;
+}
+
+/**
+ * This hook is an abstraction for keeping back press subscription
+ * logic in one place.
+ */
+export function useBackPressSubscription({
+  onBackPress,
+  isDisabled,
+}: Args): UseBackPressSubscription {
+  const [isActive, setIsActive] = React.useState(false);
+  const subscription = React.useRef<NativeEventSubscription | undefined>();
+
+  const clearSubscription = React.useCallback(
+    (shouldSetActive = true) => {
+      subscription.current?.remove();
+      subscription.current = undefined;
+      if (shouldSetActive) setIsActive(false);
+    },
+    [subscription.current]
```

Comment:
Should `current` of a `ref` be added to the deps array? I believe not, but I am not sure. (See https://stackoverflow.com/a/60476525/13006595)

Reply:
You are right